php/class-media.php

```diff
@@ -1878,34 +1887,39 @@ public function build_cached_meta( $post_id, $key, $single ) {
 	 * @param int          $post_id The attachment ID.
 	 * @param string       $key     The meta key to get.
 	 * @param string|array $data    $the meta data to update.
+	 *
+	 * @return bool
 	 */
 	public function update_post_meta( $post_id, $key, $data ) {
-		$meta_data = wp_get_attachment_metadata( $post_id, true );
-		if ( is_array( $meta_data ) ) {
-			if ( ! isset( $meta_data[ Sync::META_KEYS['cloudinary'] ] ) ) {
-				$meta_data[ Sync::META_KEYS['cloudinary'] ] = array();
-			}
-			$meta_data[ Sync::META_KEYS['cloudinary'] ][ $key ] = $data;
-			wp_update_attachment_metadata( $post_id, $meta_data );
-		} else {
-			// Update core mete data for consistency.
-			update_post_meta( $post_id, $key, $data );
+
+		$meta = $this->get_post_meta( $post_id );
+		if ( ! isset( $meta[ $key ] ) ) {
+			$meta[ $key ] = '';
 		}
+
+		if ( $meta[ $key ] !== $data ) {
+			$meta[ $key ] = $data;
+		}
+
+		return update_post_meta( $post_id, Sync::META_KEYS['cloudinary'], $meta );
 	}
 
 	/**
 	 * Delete cloudinary metadata.
 	 *
 	 * @param int    $post_id The attachment ID.
 	 * @param string $key     The meta key to get.
+	 *
+	 * @return bool
 	 */
 	public function delete_post_meta( $post_id, $key ) {
-		$meta_data = wp_get_attachment_metadata( $post_id, true );
-		if ( is_array( $meta_data ) && isset( $meta_data[ Sync::META_KEYS['cloudinary'] ] ) && is_array( $meta_data[ Sync::META_KEYS['cloudinary'] ] ) ) {
-			// Only do this side if has been set before.
-			unset( $meta_data[ Sync::META_KEYS['cloudinary'] ][ $key ] );
-			wp_update_attachment_metadata( $post_id, $meta_data );
+
+		$meta = $this->get_post_meta( $post_id );
```

Comment:
@DavidCramer 

The same here.
I think this is the reason why the meta isn't being deleted.